components/LineGraph/LineGraph.js

```diff
@@ -60,16 +60,74 @@ class LineGraph extends Component {
       .attr('class', 'bx--group-container')
       .attr('transform', `translate(${margin.left}, ${margin.top})`);
 
-    this.setState(() => {
-      return {
-        width: width - (margin.left + margin.right),
-        height: height - (margin.top + margin.bottom),
-      };
-    }, this.initialRender);
+    this.setState(
+      () => {
+        return {
+          width: width - (margin.left + margin.right),
+          height: height - (margin.top + margin.bottom),
+        };
+      },
+      () => this.initialRender()
+    );
+  }
+
+  componentWillReceiveProps(nextProps) {
+    const newState = Object.assign(this.state, nextProps);
```

Comment:
One thing here is that I think you want to have a `{}` before the this.state, should look like

```js
Object.assign({}, this.state, nextProps);
```

(basically creates a new object verses modifying `this.state`)

Reply:
I tried this and it doesn't work?